src/utils/registration-helpers.ts

```diff
@@ -760,31 +761,46 @@ export const registrationLanguageOptions = [
   getLanguageByIso6393Code('mis'),
 ];
 
+const isRegistration = (reg: Registration | RegistrationSearchItem2): reg is Registration => {
+  return (reg as Registration).entityDescription !== undefined;
+};
+
 export const registrationsHaveSamePublicationYear = (
-  reg1: Registration | RegistrationSearchItem,
-  reg2: Registration | RegistrationSearchItem
+  reg1: Registration | RegistrationSearchItem2,
+  reg2: Registration | RegistrationSearchItem2
 ) => {
-  if (!reg1.entityDescription?.publicationDate || !reg2.entityDescription?.publicationDate) {
+  const reg1PublicationYear = isRegistration(reg1)
+    ? reg1.entityDescription?.publicationDate?.year
+    : (reg1 as RegistrationSearchItem2).publicationDate?.year;
+
+  const reg2PublicationYear = isRegistration(reg2)
+    ? reg2.entityDescription?.publicationDate?.year
+    : (reg2 as RegistrationSearchItem2).publicationDate?.year;
```

Comment:
Vet ikke om jeg har ikke brukt syntaksen med `(...) is Registration` før, men likte den veldig godt! Ser ut som man ikke trenger å caste i elsen heller, siden det er gitt av originaltypen

```suggestion
    : reg1.publicationDate?.year;

  const reg2PublicationYear = isRegistration(reg2)
    ? reg2.entityDescription?.publicationDate?.year
    : reg2.publicationDate?.year;
```